bot/exts/moderation/metabase.py

```diff
@@ -0,0 +1,175 @@
+                async with self.bot.http_session.post(url, headers=self.headers, raise_for_status=True) as resp:
+                    if extension == "csv":
+                        out = await resp.text()
+                        # Save the output for use with int e
+                        self.exports[question_id] = list(csv.DictReader(StringIO(out)))
+
+                    elif extension == "json":
+                        out = await resp.json()
+                        # Save the output for use with int e
+                        self.exports[question_id] = out
+
+                        # Format it nicely for human eyes
+                        out = json.dumps(out, indent=4, sort_keys=True)
+            except ClientResponseError as e:
+                if e.status == 403:
+                    # User doesn't have access to the given question
+                    log.warning(f"Failed to auth with Metabase for question {question_id}.")
+                    await ctx.send(f":x: {ctx.author.mention} Failed to auth with Metabase for that question.")
+                else:
+                    # User credentials are invalid, or the refresh failed.
+                    # Delete the expiry time, to force a refresh on next startup.
+                    await self.session_info.delete("session_expiry")
+                    log.exception("Session token is invalid or refresh failed.")
+                    await ctx.send(f":x: {ctx.author.mention} Session token is invalid or refresh failed.")
+                return
+
+            paste_link = await send_to_paste_service(out, extension=extension)
+            if paste_link:
+                message = f":+1: {ctx.author.mention} Here's your link: {paste_link}"
+            else:
+                message = f":x: {ctx.author.mention} Link service is unavailible."
+            await ctx.send(
+                f"{message}\nYou can also access this data within internal eval by doing: "
+                f"`bot.get_cog('Metabase').exports[{question_id}]`"
+            )
+
+    # This cannot be static (must have a __func__ attribute).
+    async def cog_check(self, ctx: Context) -> bool:
+        """Only allow admins inside moderator channels to invoke the commands in this cog."""
+        checks = [
+            await has_any_role(Roles.admins).predicate(ctx),
+            is_mod_channel(ctx.channel)
+        ]
+        return all(checks)
+
+    def cog_unload(self) -> None:
+        """Cancel the init task and scheduled tasks."""
+        # It's important to wait for init_task to be cancelled before cancelling scheduled
+        # tasks. Otherwise, it's possible for _session_scheduler to schedule another task
+        # after cancel_all has finished, despite _init_task.cancel being called first.
+        # This is cause cancel() on its own doesn't block until the task is cancelled.
+        self.init_task.cancel()
+        self.init_task.add_done_callback(lambda _: self._session_scheduler.cancel_all())
```

Comment:
Shove comment into docstring.

Reply:
I agree on the first part, since the comment describes why we do this, and the caller may not necessarily care. 

As for the reason for the callback, I didn't really think about it too much, as this is a common pattern throughout the bot, such as in the Silence and Stream cogs. https://github.com/python-discord/bot/commit/2fd2c77035e87dde009c39aa7345e4871d5b41df

I'm not too familiar with the inner workings of asyncio tasks, so I can't say for certain whether this is needed or not.